form/data_transfer_objects.rst

```diff
@@ -0,0 +1,347 @@
+.. index::
+   single: Form; Data Transfer Objects
+
+How to use Data Transfer Objects (DTOs)
+=======================================
+
+Data Transfer Objects can be used by forms to separate entities from the
+validation logic of forms.
+Entities should always have a valid state.
+When entities are used as data classes for a form, the data is injected into
+the entity and validated.
+When the validation fails, the invalid data is still left in the entity.
+This can lead to invalid data being saved in the database.
+
+You will use the Maker bundle to highlight the differences between using DTOs
+and entities.
+
+.. index::
+   single: Installation
+
+Installation
+~~~~~~~~~~~~
+
+In applications using :doc:`Symfony Flex </setup/flex>`, run this command to
+install the Maker bundle before using it:
+
+.. code-block:: terminal
+
+    $ composer require maker --dev
+
+You will also need these packages in order to proceed with creating a CRUD
+example:
+
+.. code-block:: terminal
+
+    $ composer require form validator twig-bundle orm-pack security-csrf annotations
+
+Use the example ``Task`` entity from :doc:`the main forms tutorial </forms>`,
+make it a doctrine entity (this requires adding a primary key ``id``) and make
+the setters fluent.
```

Comment:
Do they have to be fluent? Not really a fan of it, maybe check out https://ocramius.github.io/blog/fluent-interfaces-are-evil/

Reply:
It's a valid point, although it should probably be discussed [in the PR](https://github.com/symfony/maker-bundle/pull/303).
In `make:entity` the code is generated with fluent setters and I think the generated code in the `Maker` bundle should resemble itself.
I'm probably still going to change it, as `make:dto` is supposed to work with existing entities and I can't assume that they have fluent interfaces.

Reply:
Since it is already in the Maker I guess it makes sense then

Reply:
I have updated this.
The example is based on the example entity from the forms docs. As there are no fluent setters, we can just omit this change.
And `make:dto` must work without fluent setters anyways - thanks for pointing me onto that.

Reply:
Glad I helped. :smile: